### DISCUSSION TITLE
Include Excalidraw/Images in the pdf

### DISCUSSION THREAD

**Discussion by hvoges** [l1xnan/obsidian-better-export-pdf#284] [Ideas]
Hello, 
I found your plugin today and fell in love with it instantly. I was looking for exactly this for some time now! 
What I am missing is the possibility to include Excalidraw images into the pdf, though. I work with Excalidraw very much and would love to integrate the created svg (or png) in the pdf without any extra work! Is this much effort?

Thank you so much!
Holger

**Comment by l1xnan** [comment]
Like the image, you can reference the Excalidraw file directly and it will display normally:
```markdown
![[your-image.png]]
![[your-excalidraw.excalidraw]]
```

**Comment by hvoges** [comment]
Yes, thats right, but therefore I must embed the image to a note first. I would like to add a picture directly without a note, because I use Obisidian for classroom-trainings, and I create a lot of them during a week. It is quite cumbersome to embed them all in a note afterwards.

------ Originalnachricht ------
Von "l1xnan" ***@***.******@***.***>>
An "l1xnan/obsidian-better-export-pdf" ***@***.******@***.***>>
Cc "Holger Voges" ***@***.******@***.***>>; "Author" ***@***.******@***.***>>
Datum 09.08.2024 11:41:57
Betreff Re: [l1xnan/obsidian-better-export-pdf] Include Excalidraw/Images in the pdf (Discussion #284)


Like the image, you can reference the Excalidraw file directly and it will display normally:

![[your-image.png]]
![[your-excalidraw.excalidraw]]

**Comment by l1xnan** [comment]
Sorry, I didn't quite understand. It sounds more like an Excalidraw usage issue. Could you give a practical example to describe your usage scenario?

**Comment by hvoges** [comment]
Yes, Sure. When I create an excalidraw-drawing in a vault, it is created as a single svg-file. When I create a pdf from a folder, only markdown-files are included. I want the svg- or the converted png-file to be part of the pdf-creation without first linking the picture in a .md-file, because I create lots of pictures.



Sorry, I didn't quite understand. It sounds more like an Excalidraw usage issue. Could you give a practical example to describe your usage scenario?

**Reply by l1xnan** [reply]
I see. I think it might be necessary to generate a virtual embedded png/svg image Markdown document to achieve this function. I will evaluate this solution in detail again. Currently, you may need to rely on some online conversion tools or Python-like scripts to solve it.
